zeppelin-server/src/main/java/com/nflabs/zeppelin/rest/InterpreterRestApi.java

```diff
@@ -0,0 +1,106 @@
+package com.nflabs.zeppelin.rest;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+
+import com.google.gson.Gson;
+import com.nflabs.zeppelin.interpreter.Interpreter;
+import com.nflabs.zeppelin.interpreter.Interpreter.RegisteredInterpreter;
+import com.nflabs.zeppelin.interpreter.InterpreterException;
+import com.nflabs.zeppelin.interpreter.InterpreterFactory;
+import com.nflabs.zeppelin.interpreter.InterpreterSetting;
+import com.nflabs.zeppelin.rest.message.NewInterpreterSettingRequest;
+import com.nflabs.zeppelin.server.JsonResponse;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Interpreter Rest API
+ *
+ */
+@Path("/interpreter")
+@Produces("application/json")
+public class InterpreterRestApi {
+  Logger logger = LoggerFactory.getLogger(InterpreterRestApi.class);
+  
+  private InterpreterFactory interpreterFactory;
+
+  Gson gson = new Gson();
+
+  public InterpreterRestApi() {
+    
+  }
+  
+  public InterpreterRestApi(InterpreterFactory interpreterFactory) {
+    this.interpreterFactory = interpreterFactory;
+  }
+
+  /**
+   * List all interpreter settings
+     * @return
+   */
+  @GET
+  @Path("setting")
```

Comment:
by convention if you do a `get /ressource` you will get all list/collection of the resources
in this case, `get /interpreter ` is what you want to do, no need to do `get /interpreter/setting`